rule.go

```diff
@@ -349,15 +353,19 @@ func (r *Rule) Evaluate(tx *Transaction) []MatchData {
 
 	// disruptive actions are only evaluated by parent rules
 	if r.ParentID == 0 {
+		matchedChainValues := matchedValues
 		// we only run the chains for the parent rule
 		for nr := r.Chain; nr != nil; {
 			tx.Waf.Logger.Debug("Evaluating rule chain", zap.Int("rule", rid), zap.String("raw", nr.Raw))
-			matchedValues = nr.Evaluate(tx)
-			if len(matchedValues) == 0 {
-				return nil
+			matchedChainValues = nr.Evaluate(tx)
+			if len(matchedChainValues) == 0 {
+				return matchedChainValues
 			}
 			nr = nr.Chain
 		}
+		if r.Chain != nil {
```

Comment:
There seem to be some problems here. You didn't add the ```matchedData``` to the ```matchedChainValues```  in the for loop. Data will be lost when merging here.

Reply:
Fixed it.